framework/src/st_things/things_stack/src/common/framework/things_data_manager.c

```diff
@@ -0,0 +1,2345 @@
+	memset(device->description, 0, (size_t) MAX_DEVICE_DESCRIPTION_LENGTH);
+
+	device->no = -1;
+	device->capa_cnt = 0;
+	device->col_cnt = 0;
+	device->sig_cnt = 0;
+	device->is_physical = 0;
+
+	for (int index = 0; index < MAX_DEVICE_CAPABILTY_CNT; index++) {
+		device->pchild_resources[index] = NULL;
+	}
+
+	return device;
+}
+
+static size_t get_json_file_size(const char *filename)
+{
+	size_t size = 0;
+	FILE *fp = fopen(filename, "r");
+
+	if (fp) {
+		if (0 == fseek(fp, 0L, SEEK_END)) {
+			size = ftell(fp);
+			rewind(fp);
+		}
+		fclose(fp);
+	}
+	return size;
+}
+
+static char *get_json_string_from_file(const char *filename)
+{
+	FILE *fp = NULL;
+	char *json_str = NULL;
+	size_t size = get_json_file_size(filename);
+	size_t readed = 0;
+	int fp_acess_cnt = 0;
+
+	if (size <= 0) {
+		THINGS_LOG_D(THINGS_ERROR, TAG, "Failed converting to JSON");
+		return NULL;
+	} else {
+		THINGS_LOG_D(THINGS_DEBUG, TAG, "File size: %d ", size);
+	}
+
+	json_str = (char *)things_malloc(size + 1);
+
+	// 1. File Read
+	fp = fopen(filename, "r");
+	if (NULL == fp) {
+		THINGS_LOG_V_ERROR(THINGS_ERROR, TAG, "Failed to open %s", filename);
+		things_free(json_str);
+		return NULL;
+	}
+	// 2. Json String read from the given file.
+	size_t bytes_read = 0;
+	while (readed < size && fp_acess_cnt < MAX_FILE_ACCESS_CNT) {
+		bytes_read = fread(&(json_str[readed]), 1, size - readed, fp);
+		readed += bytes_read;
+		THINGS_LOG_D(THINGS_DEBUG, TAG, "Read Size: %d, total Size =%d", readed, size);
+		fp_acess_cnt++;
+	}
+
+	if (fp_acess_cnt >= MAX_FILE_ACCESS_CNT) {
+		THINGS_LOG_V_ERROR(THINGS_ERROR, TAG, "Access-Times is Over for File Read. (Readed Size: %d, total Size =%d)", readed, size);
+		things_free(json_str);
+		fclose(fp);
+		return NULL;
+	}
+
+	json_str[readed] = '\0';
+
+	fclose(fp);
+	fp = NULL;
+
+	return json_str;
+}
+
+static int set_json_string_into_file(const char *filename, const char *json_str)
+{
+	FILE *fp = NULL;
+	// 1. File Read
+	fp = fopen(filename, "w");
+	if (NULL == fp) {
+		THINGS_LOG_V_ERROR(THINGS_ERROR, TAG, "Failed to open \"%s\" with Write permission.", filename);
+		return 0;
+	}
+	// 2. Json String read from the given file.
+	fwrite(json_str, 1, strlen(json_str), fp);
+
+	fclose(fp);
+	fp = NULL;
+	return 1;
+}
+
+static int init_things_info_file(cJSON *cjson, char *origin_json_str)
+{
+	int res = 1;
+
+	if (g_things_info_file_path[0] == 0) {
+		int length = 0;
+		cJSON *file_rw_path = cJSON_GetObjectItem(cjson, KEY_FILE_PATH);
+		if (NULL == file_rw_path || (length = strlen(file_rw_path->valuestring)) <= 0) {
+			THINGS_LOG_V_ERROR(THINGS_ERROR, TAG, "\"%s\" key is not exist in ocf_info file. or value is empty.", KEY_FILE_PATH);
+			return 0;
+		}
+
+		memset(g_things_info_file_path, 0, MAX_FILE_PATH_LENGTH);
+		memcpy(g_things_info_file_path, file_rw_path->valuestring, length + 1);
+		THINGS_LOG_D(THINGS_DEBUG, TAG, "OCF_INFO file path : %s", g_things_info_file_path);
+	}
+
+	FILE *fp = fopen(g_things_info_file_path, "r");
```

Comment:
It seems that code in 472~474 and 497~502 are useless.

Reply:
reflected your requirements.